app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -76,14 +76,18 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
             ViewGroup.LayoutParams.WRAP_CONTENT,
         )
 
+        val isLastGeneralOrCustomSearchEngine = state.searchEngines.filter {
+            it.isGeneral
+        }.size == 1
         state.searchEngines.filter { engine ->
             engine.type != SearchEngine.Type.APPLICATION
         }.forEach { engine ->
             val searchEngineView = makeButtonFromSearchEngine(
                 engine = engine,
                 layoutInflater = layoutInflater,
                 res = context.resources,
-                allowDeletion = state.searchEngines.size > 1,
+                allowDeletion = !(engine.isGeneral && isLastGeneralOrCustomSearchEngine) &&
+                    state.searchEngines.count { it.type != SearchEngine.Type.APPLICATION } > 1,
```

Comment:
Just a question about why we need to have an `APPLICATION` search engine available to allow deletion since `Bookmarks` / `Tabs` / `History` search engines can't be set as default?

Reply:
This is to fix a bug I caught during the implementation. The user is able to delete all the search engines from the settings list, leading to an empty list.

https://user-images.githubusercontent.com/32488956/209302478-ec9c00d4-89f6-4a76-a52f-6b002931e91a.mp4

Since this is not connected to the ticket in question here, I will create a new one and add a PR for it. Thank you for pointing it out!